### PR TITLE
Update the Code Climate image

### PR DESCRIPTION
**Pull request by neongreen**
Related to #733.

### CODE REVIEW

File: cc/Dockerfile [hunk 1]
```diff
@@ -1,46 +1,49 @@
...
 #
-FROM fpco/stack-run:lts
+FROM debian:stretch
 MAINTAINER Pat Brisbin <pbrisbin@gmail.com>
 
-ENV LANG en_US.UTF-8
+ENV LANG C.UTF-8
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends libgmp-dev
```
Comment (neongreen) [inline cc/Dockerfile]: I don't really like this part but I haven't found a haskell-env image.

File: cc/Dockerfile [hunk 2]
```diff
@@ -1,46 +1,49 @@
 #
 # Build stage
 #
-FROM fpco/stack-build:lts as builder
+FROM haskell as builder
 MAINTAINER Pat Brisbin <pbrisbin@gmail.com>
```
Comment (neongreen) [inline cc/Dockerfile]: I can put my email here or leave Pat's, it looks like we both are pretty responsive so it likely doesn't matter much.

File: cc/Dockerfile [hunk 3]
```diff
@@ -1,46 +1,49 @@
...
+RUN cabal v2-update
 
 # Install HLint
-RUN stack install --dependencies-only
+RUN cabal v2-build --dependencies-only
 COPY ./src /src/src
 COPY ./data /src/data
 COPY ./LICENSE /src/LICENSE
-RUN stack install
+COPY ./README.md /src/README.md
```
Comment (neongreen) [inline cc/Dockerfile]: This part is fragile but I wanted to avoid rewriting the Dockerfile completely. Edits welcome.

**Comment by maxjacobson** [comment]
Thanks for opening this. I'll defer to the haskell developers on the particulars. Feel free to ping me if you have any questions about how to QA this change.

**Comment by neongreen** [comment]
Not sure why CI fails – it doesn't seem to be related to this PR. Any advice?

**Comment by ndmitchell** [comment]
I have no real opinion here. If no one objects, I'll merge.

**Comment by pbrisbin** [comment]
:wave: sorry I was on vacation last week so didn't get to addressing this. I just wanted to say thanks and mention for posterity that I had no affinity for Stack here, it was just path of least resistance at the time. So if moving to cabal indeed fixings things (and avoids the bitrot/non-reproduciblility of a `stack init` in a layer), than big :+1: from me.

**Pull request merged**